File: src/main/java/com/EmployeesREST/jms/EmployeeJmsProducer.java
package com.EmployeesREST.jms;

import com.EmployeesREST.config.JmsConfig;
import com.EmployeesREST.dto.Employee;
import lombok.extern.slf4j.Slf4j;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class EmployeeJmsProducer {
    private final JmsTemplate jmsTemplate;
    public EmployeeJmsProducer(JmsTemplate jmsTemplate) {
        this.jmsTemplate = jmsTemplate;
    }

    public void addEmployee(Employee employee) {
        log.info("IN: Producer received employee: {}", employee);
        jmsTemplate.convertAndSend(JmsConfig.EMPLOYEES_QUEUE, employee);
        log.info("OUT: employee - {} - was converted and sent", employee);
    }
}

File: src/main/kotlin/com/employees/jms/EmployeeJmsProducer.kt
package com.employees.jms

import com.employees.config.JmsConfig
import com.employees.dto.Employee
import mu.KotlinLogging
import org.springframework.jms.core.JmsTemplate
import org.springframework.stereotype.Component

private val log = KotlinLogging.logger {}

@Component
class EmployeeJmsProducer(private val jmsTemplate: JmsTemplate) {
    fun addEmployee(employee: Employee?) {
        log.info("IN: Producer received employee: {}", employee)
        if (employee != null) {
            jmsTemplate.convertAndSend(JmsConfig.EMPLOYEES_QUEUE, employee)
        }
        log.info("OUT: employee - {} - was converted and sent", employee)
    }
}